### PR TITLE
Support write-only ODS for BLOB interactions

### PR DESCRIPTION
**Pull request by lagalbra**
@Microsoft/omsagent-devs 

Unit and system tests running now.
pBuild running now.
Tested locally by turning of DSC updates and ensuring success messages were printed in omsagent.log and data was sent to portal.

A PR will be created in DSC repo with new out_oms_blob.rb and bump to nxOMSPlugin version when this is approved and 2.19 has been completed.

### CODE REVIEW

**Comment by lagalbra** [comment]
Unit and system tests have passed.

File: source/code/plugins/out_oms_blob.rb [hunk 1]
```diff
@@ -136,16 +125,42 @@ def get_blob_uri(container_type, data_type, custom_data_type, suffix)
...
+    # get the blob SAS URI and committed blocks from ODS
+    # parameters
+    #   container_type: string. ContainerType of the data
+    #   data_type: string. DataTypeId of the data
+    #   custom_data_type: string. CustomDataType of the CustomLog
+    #   suffix: string. Suffix of the blob
+    # returns:
+    #   URI. blob SAS URI
+    #   string[]. a list of committed blocks
+    def get_blob_uri_and_committed_blocks(container_type, data_type, custom_data_type, suffix)
```
Comment (haitch) [inline source/code/plugins/out_oms_blob.rb]: **Size** is also available, it tells you the current size of the blob.

File: source/code/plugins/out_oms_blob.rb [hunk 2]
```diff
@@ -136,16 +125,42 @@ def get_blob_uri(container_type, data_type, custom_data_type, suffix)
...
+      return blob_uri, blocks_committed
+    end # get_blob_uri_and_committed_blocks
 
     # append data to the blob
     # parameters:
     #   uri: URI. blob URI
     #   msgs: string[]. messages
     #   file_path: string. file path
-    def append_blob(uri, msgs, file_path)
+    def append_blob(uri, msgs, file_path, blocks_committed)
```
Comment (haitch) [inline source/code/plugins/out_oms_blob.rb]: nit: I would call this blob_path.

File: source/code/plugins/out_oms_blob.rb (hunk 1 above)
Comment (lagalbra) [inline source/code/plugins/out_oms_blob.rb]: Is there any reason we'd need to use it though? I can read it from the JSON, but I don't think any of our code relies on that property.

**Comment by lagalbra** [comment]
pBuild passed.

File: source/code/plugins/out_oms_blob.rb (hunk 2 above)
Comment (lagalbra) [inline source/code/plugins/out_oms_blob.rb]: Which variable are you suggesting a change to? `uri` or `file_path`? `uri` could be considered a BLOB path, but `file_path` refers to a path on the Linux machine.

Comment (haitch) [inline source/code/plugins/out_oms_blob.rb]: nevermind, I don't see we read from this file_path, and thought it's the blob_path.  Then I see it's been used as a blob Metadata.

**Pull request merged**